app/controllers/mechanics_ride_controller.rb

```diff
@@ -0,0 +1,15 @@
+class MechanicsRideController < ApplicationController
+  def create
+    @mechanic = Mechanic.find(params[:mechanic_id])
+    @ride = Ride.find_by(id: params[:mechanic][:ride_id])
+
+    if @ride.nil?
+      flash[:notice] = "Ride not found."
+    else
+      @mechanic.rides << @ride
+      flash[:notice] = "#{@ride.name} has been added to #{@mechanic.name}'s workload."
```

Comment:
Good use of a flash message in both sad & happy paths, but this functionality is not tested in a feature test. Try to focus on TDD to build out one bit of functionality at a time, and then implement it after the test is written. Believe me, TDD pays off in the end and it _never goes away_!